yt_dlp/compat.py

```diff
@@ -2982,6 +2984,23 @@ def unpack(self, string):
         compat_Struct = struct.Struct
 
 
+try:
+    from tempfile import TemporaryDirectory as compat_TemporaryDirectory
+except ImportError:
+    class compat_TemporaryDirectory:
+        def __init__(self, prefix=None):
+            self._prefix = prefix
+            self._path = None
+
+        def __enter__(self):
+            self._path = tempfile.mkdtemp(prefix=self._prefix)
+            return self._path
+
+        def __exit__(self, exc_type, exc_val, exc_tb):
+            if self._path is not None:
+                shutil.rmtree(self._path)
```

Comment:
no, it isn't. I left it because of what you said earlier about why yt-dlp still uses compat.py but since this is a new feature I can remove it and use tempfile directly

Reply:
I've also actually removed most of the code from `compat.py` now